diff --git a/cpp/src/gandiva/precompiled/CMakeLists.txt b/cpp/src/gandiva/precompiled/CMakeLists.txt
index d7c7ef157..2c92e28d2 100644
--- a/cpp/src/gandiva/precompiled/CMakeLists.txt
+++ b/cpp/src/gandiva/precompiled/CMakeLists.txt
@@ -49,6 +49,14 @@ else()
   set(PLATFORM_CLANG_OPTIONS -std=c++17)
 endif()
 
+if (USE_LLVM_CXX)
+  message(STATUS "USE LLVM CXX")
+  set(LLVM_CXX_INCLUDE -I${LLVM_INCLUDE_DIRS}/c++/v1 -I${LLVM_INCLUDE_DIRS}/x86_64-unknown-linux-gnu/c++/v1)
+else()
+  message(STATUS "NOT USE LLVM CXX")
+  set(LLVM_CXX_INCLUDE "")
+endif()
+
 # Create bitcode for each of the source files.
 foreach(SRC_FILE ${PRECOMPILED_SRCS})
   get_filename_component(SRC_BASE ${SRC_FILE} NAME_WE)
@@ -79,6 +87,7 @@ foreach(SRC_FILE ${PRECOMPILED_SRCS})
        -o
        ${BC_FILE}
        ${ARROW_GANDIVA_PC_CXX_FLAGS}
+       ${LLVM_CXX_INCLUDE}
        -I${CMAKE_SOURCE_DIR}/src
        -I${ARROW_BINARY_DIR}/src)
 
