--- a/configure.ac	2024-11-29 14:13:59.817559244 +0800
+++ b/configure.ac	2024-11-29 14:14:16.668261779 +0800
@@ -44,7 +44,7 @@
                  color-tests parallel-tests
                  silent-rules]
                  m4_bmatch(m4_defn([AC_PACKAGE_VERSION]), [[-_]],
-                           [gnu], [gnits]))
+                           [gnu], [gnu]))
 AM_SILENT_RULES([yes])
 AC_CONFIG_HEADERS([lib/config.h:lib/config.in.h])
 
