--- CMakeLists.txt	2024-01-05 14:59:42.233660898 +0800
+++ src/CMakeLists.txt	2024-01-08 14:07:39.058107005 +0800
@@ -153,8 +153,8 @@
     endif( )
     if( NOT USE_SYSTEM_TZ_DB AND NOT MANUAL_TZ_DB )
         find_package( CURL REQUIRED )
-        target_include_directories( date-tz SYSTEM PRIVATE ${CURL_INCLUDE_DIRS} )
-        target_link_libraries( date-tz PRIVATE ${CURL_LIBRARIES} )
+	    target_include_directories( date-tz SYSTEM PRIVATE ${CURL_DIR}/../../../include )
+	    target_link_libraries( date-tz PRIVATE ${CURL_DIR}/../../../lib )
     endif( )
 endif( )
 
