diff -ur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2022-12-06 14:15:51.000000000 +0800
+++ b/CMakeLists.txt	2023-09-14 10:57:03.633564428 +0800
@@ -216,54 +216,11 @@
   set(SUN TRUE)
 endif()

-find_package(Git)
-if(Git_FOUND)
-  if (NOT DEFINED GIT_COMMIT_HASH)
-    execute_process(
-            COMMAND ${GIT_EXECUTABLE} log -1 --format=%h
-            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-            RESULT_VARIABLE GIT_RESULT
-            OUTPUT_VARIABLE GIT_COMMIT_HASH
-            OUTPUT_STRIP_TRAILING_WHITESPACE)
-  endif()
-  execute_process(
-          COMMAND ${GIT_EXECUTABLE} describe --tags --abbrev=0
-          WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-          OUTPUT_VARIABLE GIT_LAST_TAG
-          OUTPUT_STRIP_TRAILING_WHITESPACE)
-  execute_process(
-          COMMAND ${GIT_EXECUTABLE} describe --tags --long
-          WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-          OUTPUT_VARIABLE GIT_ITERATION
-          OUTPUT_STRIP_TRAILING_WHITESPACE)
-else()
-  message("Git NOT FOUND")
-endif()
-
-if(GIT_RESULT EQUAL "0")
-  string(REGEX REPLACE "v([0-9]+).[0-9]+.[0-9]+" "\\1" DUCKDB_MAJOR_VERSION "${GIT_LAST_TAG}")
-  string(REGEX REPLACE "v[0-9]+.([0-9]+).[0-9]+" "\\1" DUCKDB_MINOR_VERSION "${GIT_LAST_TAG}")
-  string(REGEX REPLACE "v[0-9]+.[0-9]+.([0-9]+)" "\\1" DUCKDB_PATCH_VERSION "${GIT_LAST_TAG}")
-  string(REGEX REPLACE ".*-([0-9]+)-.*" "\\1" DUCKDB_DEV_ITERATION "${GIT_ITERATION}")
-
-  if(DUCKDB_DEV_ITERATION EQUAL 0)
-    # on a tag; directly use the version
-    set(DUCKDB_VERSION "${GIT_LAST_TAG}")
-  else()
-    # not on a tag, increment the patch version by one and add a -devX suffix
-    math(EXPR DUCKDB_PATCH_VERSION "${DUCKDB_PATCH_VERSION}+1")
-    set(DUCKDB_VERSION "v${DUCKDB_MAJOR_VERSION}.${DUCKDB_MINOR_VERSION}.${DUCKDB_PATCH_VERSION}-dev${DUCKDB_DEV_ITERATION}")
-  endif()
-else()
-  # fallback for when building from tarball
-  set(DUCKDB_MAJOR_VERSION 0)
-  set(DUCKDB_MINOR_VERSION 0)
-  set(DUCKDB_PATCH_VERSION 1)
-  set(DUCKDB_DEV_ITERATION 0)
-  set(DUCKDB_VERSION "v${DUCKDB_MAJOR_VERSION}.${DUCKDB_MINOR_VERSION}.${DUCKDB_PATCH_VERSION}-dev${DUCKDB_DEV_ITERATION}")
-endif()
-
-message(STATUS "git hash ${GIT_COMMIT_HASH}, version ${DUCKDB_VERSION}")
+set(DUCKDB_MAJOR_VERSION 0)
+set(DUCKDB_MINOR_VERSION 0)
+set(DUCKDB_PATCH_VERSION 1)
+set(DUCKDB_DEV_ITERATION 0)
+set(DUCKDB_VERSION "v${DUCKDB_MAJOR_VERSION}.${DUCKDB_MINOR_VERSION}.${DUCKDB_PATCH_VERSION}-dev${DUCKDB_DEV_ITERATION}")

 option(AMALGAMATION_BUILD
        "Build from the amalgamation files, rather than from the normal sources."
diff -ur a/scripts/amalgamation.py b/scripts/amalgamation.py
--- a/scripts/amalgamation.py	2022-12-06 14:15:51.000000000 +0800
+++ b/scripts/amalgamation.py	2023-09-14 10:55:43.448810633 +0800
@@ -257,8 +257,8 @@
         hfile.write("#define DUCKDB_AMALGAMATION 1\n")
         if extended_amalgamation:
             hfile.write("#define DUCKDB_AMALGAMATION_EXTENDED 1\n")
-        hfile.write("#define DUCKDB_SOURCE_ID \"%s\"\n" % git_commit_hash())
-        hfile.write("#define DUCKDB_VERSION \"%s\"\n" % git_dev_version())
+#        hfile.write("#define DUCKDB_SOURCE_ID \"%s\"\n" % git_commit_hash())
+#        hfile.write("#define DUCKDB_VERSION \"%s\"\n" % git_dev_version())

         for fpath in main_header_files:
             hfile.write(write_file(fpath))
diff -ur a/third_party/thrift/thrift/TApplicationException.h b/third_party/thrift/thrift/TApplicationException.h
--- a/third_party/thrift/thrift/TApplicationException.h 2022-12-06 14:15:51.000000000 +0800
+++ b/third_party/thrift/thrift/TApplicationException.h 2023-12-17 17:10:08.964516203 +0800
@@ -20,6 +20,7 @@
 #ifndef _DUCKDB_THRIFT_TAPPLICATIONEXCEPTION_H_
 #define _DUCKDB_THRIFT_TAPPLICATIONEXCEPTION_H_ 1

+#include <cstdint>
 #include "thrift/Thrift.h"

 namespace duckdb_apache {
