diff -ur a/scripts/amalgamation.py b/scripts/amalgamation.py
--- a/scripts/amalgamation.py	2023-01-19 16:03:55.119686874 +0800
+++ b/scripts/amalgamation.py	2023-01-19 16:03:39.680710636 +0800
@@ -257,8 +257,8 @@
         hfile.write("#define DUCKDB_AMALGAMATION 1\n")
         if extended_amalgamation:
             hfile.write("#define DUCKDB_AMALGAMATION_EXTENDED 1\n")
-        hfile.write("#define DUCKDB_SOURCE_ID \"%s\"\n" % git_commit_hash())
-        hfile.write("#define DUCKDB_VERSION \"%s\"\n" % git_dev_version())
+#        hfile.write("#define DUCKDB_SOURCE_ID \"%s\"\n" % git_commit_hash())
+#        hfile.write("#define DUCKDB_VERSION \"%s\"\n" % git_dev_version())

         for fpath in main_header_files:
             hfile.write(write_file(fpath))
