diff -ur a/thrift/lib/cpp2/async/HeaderServerChannel.h b/thrift/lib/cpp2/async/HeaderServerChannel.h
--- a/thrift/lib/cpp2/async/HeaderServerChannel.h	2021-11-29 14:14:12.000000000 +0800
+++ b/thrift/lib/cpp2/async/HeaderServerChannel.h	2022-10-13 16:22:03.142090462 +0800
@@ -108,7 +108,7 @@
     }

     bool isOneway() const override {
-      return header_->getSequenceNumber() == ONEWAY_REQUEST_ID;
+      return header_ && header_->getSequenceNumber() == ONEWAY_REQUEST_ID;
     }

     bool includeEnvelope() const override { return true; }
diff -ur a/thrift/lib/cpp2/server/IOWorkerContext.h b/thrift/lib/cpp2/server/IOWorkerContext.h
--- a/thrift/lib/cpp2/server/IOWorkerContext.h	2022-12-26 14:14:12.000000000 +0800
+++ b/thrift/lib/cpp2/server/IOWorkerContext.h	2023-01-06 10:19:57.688867465 +0800
@@ -64,7 +64,7 @@
           auto aliveLocked = alive->rlock();
           if (*aliveLocked) {
             // IOWorkerContext is still alive and so is replyQueue_
-            queue->startConsumingInternal(&evb);
+            queue->startConsuming(&evb);
           }
         });
   }
diff -ur a/thrift/compiler/generate/json.cc b/thrift/compiler/generate/json.cc
--- a/thrift/compiler/generate/json.cc  2022-12-25 17:00:33.000000000 +0800
+++ b/thrift/compiler/generate/json.cc  2023-12-12 10:57:27.256511493 +0800
@@ -16,6 +16,7 @@

 #include <thrift/compiler/generate/json.h>

+#include <cstdint>
 #include <ostream>
 #include <sstream>
