diff -ur a/CMake/FindLibDwarf.cmake b/CMake/FindLibDwarf.cmake
--- a/CMake/FindLibDwarf.cmake	2022-12-27 15:59:36.511072945 +0800
+++ b/CMake/FindLibDwarf.cmake	2022-12-27 15:59:59.001018283 +0800
@@ -15,7 +15,7 @@
 # dwarf.h is typically installed in a libdwarf/ subdirectory on Debian-style
 # Linux distributions.  It is not installed in a libdwarf/ subdirectory on Mac
 # systems when installed with Homebrew.  Search for it in both locations.
-find_path(LIBDWARF_INCLUDE_DIR NAMES dwarf.h PATH_SUFFIXES libdwarf)
+find_path(LIBDWARF_INCLUDE_DIR NAMES dwarf.h PATH_SUFFIXES libdwarf libdwarf-0)
 mark_as_advanced(LIBDWARF_INCLUDE_DIR)

 find_library(LIBDWARF_LIBRARY NAMES dwarf)
diff -ur a/folly/experimental/exception_tracer/ExceptionTracerLib.cpp b/folly/experimental/exception_tracer/ExceptionTracerLib.cpp
--- a/folly/experimental/exception_tracer/ExceptionTracerLib.cpp	2022-12-24 20:17:25.000000000 +0800
+++ b/folly/experimental/exception_tracer/ExceptionTracerLib.cpp	2022-12-27 16:25:58.444339235 +0800
@@ -98,6 +98,7 @@
 #define __builtin_unreachable()
 #endif

+#if 0
 namespace __cxxabiv1 {

 void __cxa_throw(
@@ -156,5 +157,6 @@
 }

 } // namespace std
+#endif

 #endif // defined(__GLIBCXX__)
diff -ur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -77,10 +77,14 @@ set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)

 if("${CMAKE_LIBRARY_ARCHITECTURE}" STREQUAL "")
-  # CMAKE_LIBRARY_ARCHITECTURE is not always set, so we have to assume
-  # arch might be x86_64
-  message(WARNING "CMAKE_LIBRARY_ARCHITECTURE not set, assuming x86_64")
-  set(IS_X86_64_ARCH ON)
+  if(NOT "${CMAKE_HOST_SYSTEM_PROCESSOR}" STREQUAL "x86_64")
+    set(IS_X86_64_ARCH OFF)
+  else()
+    # CMAKE_LIBRARY_ARCHITECTURE is not always set, so we have to assume
+    # arch might be x86_64
+    message(WARNING "CMAKE_LIBRARY_ARCHITECTURE not set, assuming x86_64")
+    set(IS_X86_64_ARCH ON)
+  endif()
 else()
   string(FIND "${CMAKE_LIBRARY_ARCHITECTURE}" "x86_64" IS_X86_64_ARCH)
   if(IS_X86_64_ARCH STREQUAL "-1")
diff -ur a/folly/system/AtFork.cpp b/folly/system/AtFork.cpp
--- a/folly/system/AtFork.cpp   2022-12-24 20:17:25.000000000 +0800
+++ b/folly/system/AtFork.cpp   2023-12-12 10:44:30.560711808 +0800
@@ -16,6 +16,9 @@

 #include <folly/system/AtFork.h>

+#include <stdexcept>
+#include <system_error>
+
 #include <folly/ScopeGuard.h>
 #include <folly/lang/Exception.h>
 #include <folly/portability/PThread.h>
